Action 'Bump version/Update Milestone' is failing to update milestones

https://github.com/checkstyle/checkstyle/actions/runs/3356052486/jobs/5560809102

```
Run .ci/update-github-milestone.sh 10.4
Error: Define GITHUB_TOKEN environment variable
Error: Process completed with exit code 1.
```

but commit is pushed - https://github.com/checkstyle/checkstyle/commit/9b4e1a6eec674ca1108f6aaa64f3616532718c98 ,  but milestone is failed to be updated

commit `minor: Bump release version to 10.4` was very confusing to me, we should rename it be `minor: Bump release version to 10.4-SNAPSHOT`

--- a/.github/workflows/bump-version-and-update-milestone.yml
+++ b/.github/workflows/bump-version-and-update-milestone.yml
@@ -32,10 +32,10 @@ jobs:
         run: |
           git config user.name 'github-actions[bot]'
           git config user.email 'github-actions[bot]@users.noreply.github.com'
-          git commit -am "minor: Bump release version to ${{ github.event.inputs.version }}"
+          git commit -am "minor: Bump version to ${{ github.event.inputs.version }}-SNAPSHOT"
           git push
       - name: GitHub Milestone
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
         run: |
           .ci/update-github-milestone.sh ${{ github.event.inputs.version }}
-        env:
-          READ_ONLY_TOKEN: ${{ github.token }}

